packages/server/src/fhir/operations/projectinit.ts

```diff
@@ -118,16 +117,16 @@ export async function projectInitHandler(req: Request, res: Response): Promise<v
  * Creates a new project.
  * @param projectName - The new project name.
  * @param admin - The Project admin user.
- * @returns The new project, admin, membership and associated data.
+ * @returns The new project, and associated data.  Profile and membership are returned if and only if an admin user is specified.
  */
 export async function createProject(
   projectName: string,
-  admin: User
+  admin?: User
 ): Promise<{
   project: Project;
-  profile: ProfileResource;
-  membership: ProjectMembership;
   client: ClientApplication;
+  profile?: ProfileResource;
```

Comment:
If you wanted to get fancy / abuse TypeScript, you could make the return type conditional on the `admin` param.

See packages/server/src/test.setup.ts for an example

Something like:

```ts
export type CreateProjectResult<T extends User | undefined> = {
  project: Project;
  client: ClientApplication;
  profile: T extends User ? ProfileResource : undefined;
  membership: T extends User ? ProjectMembership : undefined;
};

export async function createProject<T extends User | undefined>(
  projectName: string,
  admin: T
): Promise<CreateProjectResult<T>> {
```

Totally optional.  It might clean up some of the casts that you had to add.

Reply:
Ooooh nice!  I had attempted something similar initially, but didn't find this correct incantation